tanner/tests/test_template_injection.py

```diff
@@ -0,0 +1,50 @@
+import asyncio
+import unittest
+
+from tanner.emulators.template_injection import TemplateInjection
+
+
+class TestTemplateInjection(unittest.TestCase):
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(None)
+        self.handler = TemplateInjection(loop=self.loop)
+        self.result = None
+        self.expected_result = None
+        self.returned_result = None
+
+    def test_scan(self):
+        payload = '{{config}}'
+
+        self.expected_result = dict(name='template_injection', order=3)
+        self.returned_result = self.handler.scan(payload)
+        self.assertEqual(self.returned_result, self.expected_result)
+
+    def test_scan_negative(self):
+        payload = '{7*7}'
+
+        self.expected_result = None
+        self.returned_result = self.handler.scan(payload)
+        self.assertEqual(self.returned_result, self.expected_result)
+
+    def test_handle_mako(self):
+        payload = "<%\nimport os\nx=os.popen('id').read()\n%>${x}"
+
+        self.returned_result = self.handler.get_injection_result(payload)
+        self.expected_result = 'uid=0(root) gid=0(root) groups=0(root)'
+        self.assertIn(self.expected_result, self.returned_result)
+
+    def test_handle_jinja2(self):
+        payload = '{{ 7*"7" }}'
+
+        self.returned_result = self.handler.get_injection_result(payload)
+        self.expected_result = '7777777'
+        self.assertIn(self.expected_result, self.returned_result)
+
+    def test_handle_tornado(self):
```

Comment:
So this can compromise can the entire honeypot and attack will soon understand that it is talking to a honeypot. Just think about why we're using Docker containers / phpox for `code execution` emulators.

Reply:
I am just demonstrating the possibilities. In real production, all of this can be access controlled and also suitable filters can be applied to prevent such payloads.

Reply:
I think you don't understand honeypot, its aim is to collect all sort of working payloads (simple+complex).

Reply:
The emulator should emulate almost all the payloads.